src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -939,7 +944,6 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (_ *
 	}
 	s.mu.Unlock()
 
-	span.End()
 	katatrace.StopTracing(s.rootCtx)
```

Comment:
Should we also defer `katatrace.StopTracing(s.rootCtx)` to make sure that it executes in the case described in #2930?

Reply:
hi @cmaf , I am not sure that if the `shutdown ` action exec failed, we should still stop the root context span or not. So I keep these business codes, only the main process correctly executed, we stop the root context span. Can we have a try to execute the `shutdown` action again, if the `shutdown` commands executed failed?